pkg/scanner/scanner.go

```diff
@@ -239,41 +294,99 @@ func (s *Scanner) Results() *types.ScanResults {
 	}
 }
 
-func (s *Scanner) HandleResult(result *forwarding.ImageVulnerabilities) error {
+func (s *Scanner) shouldIgnoreResult(scanD *scanData, resultScanUUID string, image string) bool {
+	if scanD == nil {
+		log.WithFields(s.logFields).Warnf("no scan data for image '%v', probably an old scan result - ignoring", image)
+		return true
+	}
+
+	if resultScanUUID != scanD.scanUUID {
+		log.WithFields(s.logFields).Warnf("Scan UUID mismatch, probably an old scan result - ignoring. image=%v, received=%v, expected=%v", image, resultScanUUID, scanD.scanUUID)
+		return true
+	}
+
+	if scanD.timeout {
+		log.WithFields(s.logFields).Warnf("Scan result after timeout - ignoring. image=%v, scan uuid=%v", image, resultScanUUID)
+		return true
+	}
+
+	if scanD.completed {
+		log.WithFields(s.logFields).Warnf("Duplicate result for image scan. image=%v, scan uuid=%v", image, resultScanUUID)
+		return true
+	}
+
+	return false
+}
+
+func (s *Scanner) HandleVulnerabilitiesResult(result *forwarding.ImageVulnerabilities) error {
 	s.Lock()
 	defer s.Unlock()
 
 	scanD, ok := s.imageToScanData[result.Image]
-	if !ok || scanD == nil {
-		log.WithFields(s.logFields).Warnf("no scan data for image '%v', probably an old scan result - ignoring", result.Image)
+	if !ok || s.shouldIgnoreResult(scanD, result.ScanUUID, result.Image) {
```

Comment:
It seems that 
```
	scanD, ok := s.imageToScanData[result.Image]
	if !ok || scanD == nil {		if !ok || s.shouldIgnoreResult(scanD, result.ScanUUID, result.Image) {
		log.WithFields(s.logFields).Warnf("no scan data for image '%v', probably an old scan result - ignoring", result.Image)			log.WithFields(s.logFields).Warnf("Ignoring vulnerabilities result for image '%v'", result.Image)
		return nil			return nil
	}		}
```
And 
```
	if !scanD.completed {
		log.WithFields(s.logFields).Infof("Total scan is not yet completed for image %v", result.Image)
		return nil			return nil
	}		}


	select {
	case scanD.resultChan <- true:
	default:
		log.WithFields(s.logFields).Warnf("Failed to notify upon received result scan. image=%v, scan-uuid=%v", result.Image, result.ScanUUID)
	}

	return nil
}
```

should be in a common part of the handling results and the rest should be in a separate function